diff --git a/node_modules/typedarray-pool/pool.js b/node_modules/typedarray-pool/pool.js
index 8d3a202..18c3a90 100644
--- a/node_modules/typedarray-pool/pool.js
+++ b/node_modules/typedarray-pool/pool.js
@@ -2,9 +2,10 @@
 
 var bits = require('bit-twiddle')
 var dup = require('dup')
-var Buffer = require('buffer').Buffer
+// const { Buffer } = require('node:buffer')
 
 //Legacy pool support
+var global = global || self;
 if(!global.__TYPEDARRAY_POOL) {
   global.__TYPEDARRAY_POOL = {
       UINT8     : dup([32, 0])
@@ -47,7 +48,7 @@ var DATA    = POOL.DATA
   , BUFFER  = POOL.BUFFER
 
 exports.free = function free(array) {
-  if(Buffer.isBuffer(array)) {
+  if(ethereumjs.Buffer.Buffer.isBuffer(array)) {
     BUFFER[bits.log2(array.length)].push(array)
   } else {
     if(Object.prototype.toString.call(array) !== '[object ArrayBuffer]') {
@@ -228,7 +229,8 @@ function mallocBuffer(n) {
   if(cache.length > 0) {
     return cache.pop()
   }
-  return new Buffer(n)
+  //return new Buffer(n)
+  return new ethereumjs.Buffer.Buffer(n)
 }
 exports.mallocBuffer = mallocBuffer
 
